flocker/control/_model.py

```diff
@@ -463,7 +478,7 @@ def move_application(self, application, target_node):
                 if container.name == application.name:
                     # We only need to perform a move if the node currently
                     # hosting the container is not the node it's moving to.
-                    if node.hostname != target_node.hostname:
+                    if node.uuid != target_node.uuid:
```

Comment:
:question:

A `node.same_node_as(target_node)` API would be a good way to isolate this comparison logic and avoid duplicating it throughout the codebase.